server/src/test/java/io/crate/planner/SubQueryPlannerTest.java

```diff
@@ -233,4 +235,47 @@ public void testJoinWithAggregationOnSubSelectsWithAggregations() throws Excepti
             exactlyInstanceOf(GroupProjection.class)
         );
     }
+
+
+    @Test
+    public void test_filter_on_top_of_multi_phase_is_pushed_down_if_possible() throws Exception {
+        e.addView(
+            new RelationName("doc", "v1"),
+            "select * from doc.t1 where a in (select b from doc.t2)"
+        );
+        LogicalPlan plan1 = e.logicalPlan("select * from v1 where a = 'foo'");
+        assertThat(plan1).as("Is pushed down if not using sub-query output").hasOperators(
+            "Rename[a, x, i] AS doc.v1",
+            "  └ MultiPhase",
+            "    └ Collect[doc.t1 | [a, x, i] | ((a = 'foo') AND (a = ANY((SELECT b FROM (doc.t2)))))]",
+            "    └ OrderBy[b ASC]",
+            "      └ Collect[doc.t2 | [b] | true]"
+        );
+
+        e.addView(
+            new RelationName("doc", "v2"),
+            "select *, (select mountain from sys.summits limit 1) as mountain from doc.t1"
+        );
+        LogicalPlan plan2 = e.logicalPlan("select * from v2 where mountain = 'Brandstein'");
```

Comment:
Shouldn't this work as well? so to push down filters inside the multiphase subquery?

Reply:
I don't think that works.

```
select *, (select mountain from sys.summits where mountain = 'foo' limit 1) as mountain from doc.t1;
```

Would return a row where `mountain` is `null` instead of returning no rows.
Compare with:

```
select * from (select *, (select mountain from sys.summits limit 1) as mountain from doc.t1) t where mountain = 'foo';
```